### ISSUE TITLE
ccPointCloud::createNewCloudFromVisibilitySelection does not always create a new point cloud

### ISSUE THREAD

**Issue by maz-1** [CloudCompare/CloudCompare#2078]
if selection table selects all points in the point cloud, `ccPointCloud::createNewCloudFromVisibilitySelection` will just return the original point cloud reference, which is confusing since this function does not always create a new point cloud. Maybe we should add a new fuction with a different name like `createNewCloudFromVisibilitySelectionIfPatial` and change the default behaviour of existing function.
If changing APIs is not acceptable, at least we should update the documentation to warn developers about this.

**Comment by dgirardeau** [comment]
Yes, this is done on purpose to avoid duplicating the point cloud for no good reason (for the sake of memory, when working with very large point clouds).

Feel free to propose an additional method (or simply a boolean to force the creation of a new point cloud).
